### PR TITLE
Determine the OS we're running on

### PR DESCRIPTION
**Pull request by adrcunha**
Set IS_LINUX, IS_OSX and IS_WINDOWS accordingly.

### CODE REVIEW

File: scripts/library.sh [hunk 1]
```diff
@@ -36,6 +36,20 @@ readonly IS_PROW
...
 
+# Useful flags about the current OS
+IS_LINUX=0
+IS_OSX=0
+IS_WINDOWS=0
+case "${OSTYPE}" in
+  darwin*) IS_OSX=1 ;;
+  linux*) IS_LINUX=1 ;;
+  msys*) IS_WINDOWS=1 ;;
+  *) echo "** Internal error in library.sh, unknown OS" ; exit 1 ;;
```
Comment (adrcunha) [inline scripts/library.sh]:
```suggestion
  *) echo "** Internal error in library.sh, unknown OS '${OSTYPE}'" ; exit 1 ;;
```

File: scripts/library.sh [hunk 2]
```diff
@@ -36,6 +36,20 @@ readonly IS_PROW
 readonly REPO_ROOT_DIR="$(git rev-parse --show-toplevel)"
 readonly REPO_NAME="$(basename ${REPO_ROOT_DIR})"
 
+# Useful flags about the current OS
+IS_LINUX=0
+IS_OSX=0
+IS_WINDOWS=0
+case "${OSTYPE}" in
```
Comment (chaodaiG) [inline scripts/library.sh]: What would the value of `${OSTYPE}` be for a linux container running on OSX?

Comment (adrcunha) [reply scripts/library.sh]: I guess it would be Linux, because it's bash who defines OSTYPE.

Comment (chaodaiG) [reply scripts/library.sh]: should we document it here or would it be the caller's job to figure this out?

Comment (adrcunha) [reply scripts/library.sh]: This is a corner case and I can't imagine such situation in the present state of test-infra. If we ever get to this situation, I'm sure way more issues will arise when running these scripts (assuming they will be even being used then) than realizing that for some reason bash wrongly sets OSTYPE.

Comment (chaodaiG) [reply scripts/library.sh]: make sense
